src/recon.rs

```diff
@@ -2458,7 +2458,7 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                         * (f.b4_stride >> 1)
                         + ((t.b.x >> 1) + (t.b.y & 1)) as isize)
                         as isize;
-                    f.frame_thread.pal.as_slice::<BD>()[index as usize][0].as_ptr()
+                    f.frame_thread.pal.index::<BD>(index as usize)[0].as_ptr()
```

Comment:
Do we need to store the guard on the stack here? This looks like we're taking a pointer to data in a `DisjointMut`, so I think we need to have a guard on the stack while that pointer is being used.

Reply:
Good point, done.